src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdValidationConfiguration.cs

```diff
@@ -78,6 +79,11 @@ public void Configure(AuthenticationOptions options)
             // this using block won't be disposed until the service scope itself is released.
             using (var scope = CreateTenantScope(settings.Tenant))
             {
+                if (scope == null)
```

Comment:
Doing that is problematic as it will leave the tenant that uses the validation feature in a state where it can't properly validate incoming tokens.

If this is the pattern we adopt, we'll probably have to replace the `return` by a signal telling the module middleware to "restart" the tenant (by creating a new `ShellContext` with a new service provider). E.g something like `throw new ShellFatalException()`.

Reply:
As said below, i think the only use case is when the tenant is already disabled and this is what i'm ckecking in `CreateTenantScope()`. But i will remove the check in `CreateTenantScope()` because currently i do it directly in `EnterServiceScope()` which returns null if the shell is disabled.

When we are in an active scope of a given tenant that's ok. But here the singularity is that we create a scope from another tenant, so at some point we need to check if it is not disabled.

So, maybe not a pattern to adopt, just a temporary one. But it depends on what we want to do in such a case. Hmm, maybe we can't throw a non catched exception if the result is that we can't restart the application.

As a side note, doing some tests on `OpenId`, i'm a newbie around this ;), i was in some cases where i was not able to restart the application, only by doing a new setup. This when it was telling to me that there is a missing certificate, and also when there was a duplicate certificate.